### PR TITLE
Fix to #38693 (minimal UT)

### PR DESCRIPTION
**Pull request by jczaja**
### PR types
Bug fixes 

### PR changes
 Others 

### Describe
Fix to https://github.com/PaddlePaddle/Paddle/issues/38693 . This PR is the same functionality as #40923 but with stripped down UT due to timeout issues.

### CODE REVIEW

**Comment by jczaja** [comment]
@Silv3S , @tsocha Please review

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv3d_op.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        data_format=st.sampled_from(["NCDHW", "NDHWC"]),
+        dilations=st.sampled_from([[1, 2, 1]]),
+        padding_algorithm=st.sampled_from(["EXPLICIT"]),
+        groups=st.sampled_from([2]),
+        paddings=st.sampled_from([[0, 3, 2]]),
+        strides=st.sampled_from([[1, 2, 1]]),
+        batch_size=st.integers(
+            min_value=1, max_value=4), )
+    def test(self, *args, **kwargs):
+        self.run_test(*args, **kwargs)
```
Comment (jakpiase) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv3d_op.py:88]: Is executing this test still taking longer than an hour?

Comment (jczaja) [reply python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv3d_op.py:88]: not anymore. I just reduced it to minimum and expecting @baoachun (original author) to extended if needed. Full UT that I was asked to test (https://github.com/PaddlePaddle/Paddle/pull/38692) takes over ~4100 seconds.

**Review by jakpiase — DISMISSED** [review]
LGTM

File: paddle/phi/common/layout.h [hunk 2]
```diff
@@ -26,13 +26,13 @@ enum class DataLayout {
...
   NCHW,
+  NCDHW,
+  NDHWC,
   MKLDNN,
   SPARSE_COO,
   SPARSE_CSR,
   PSTRING_UNION,
   NUM_DATA_LAYOUTS,
-  NDHWC,
-  NCDHW,
```
Comment (tsocha) [inline paddle/phi/common/layout.h:35]: Why did you have to change an order of these layouts?

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv3d_op.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+from program_config import TensorConfig, ProgramConfig, OpConfig
+import numpy as np
+import paddle.inference as paddle_infer
+from functools import partial
+from typing import Optional, List, Callable, Dict, Any, Set
+import unittest
+
+import hypothesis
+from hypothesis import given, settings, seed, example, assume
+import hypothesis.strategies as st
```
Comment (tsocha) [inline python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv3d_op.py:25]: There is a lot of imports, could you verify which are needed?

**Review by tsocha — DISMISSED** [review]

**Review by Silv3S — DISMISSED** [review]
LGTM

**Comment by jczaja** [comment]
@XieYunshen Could you please approve changes for setting of timeout for newly added test?

File: python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv3d_op.py (hunk 3 above)
Comment (jczaja) [reply python/paddle/fluid/tests/unittests/ir/inference/test_mkldnn_conv3d_op.py:25]: As discussed above I incorporated this UT from PR #38692 and I hope @baoachun will fine-tune this UT in future PR also taking into account you suggestions.

**Comment by jczaja** [comment]
@Aganlengzi Could you please help in getting PR-CI-APPROVAL pass?

**Review by lidanqing-vv — DISMISSED** [review]
LGTM

**Review by baoachun — DISMISSED** [review]
LGTM

**Comment by lidanqing-vv** [comment]
@baoachun Hi could you please review if this UT is ok?

**Comment by Aganlengzi** [comment]
> @Aganlengzi Could you please help in getting PR-CI-APPROVAL pass?

hi, @jczaja 
for CheckPRTemplate, you can try re-edit the description(edit and save),
for PR-CI-APPROVAL, @XieYunshen please help review, then you can re-run the PR-CI-APPROVAL.

**Comment by lidanqing-vv** [comment]
@baoachun Hi, this PR need to be merged or cherry-pick into 2.3RC right? I will rerun all CIs cause it passed 7 days, and then will need Baidu guy's approval.

**Comment by jczaja** [comment]
@chalsliu Could you please approve changes([python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt](https://github.com/PaddlePaddle/Paddle/pull/41026/files#diff-873f20a3de6db44a3dbc57fb79df3cd12f2ce0837e3e3c7ae50420dbb7f62d4b)) for setting of timeout for newly added test? This is needed to have PR-CI-APPROVAL passed

**Review by XieYunshen — DISMISSED** [review]
LGTM

**Comment by XieYunshen** [comment]
> @XieYunshen Could you please approve changes for setting of timeout for newly added test?

done

**Comment by jczaja** [comment]
@Aganlengzi  Couple you please help in explaining why PR-CE-Framework and PR-CI-iScan-Python are failing . I have rerun them couple oftimes without success. Please advice

**Comment by baoachun** [comment]
Hi @jczaja, Can you merge the latest code and submit it again?

**Comment by lidanqing-vv** [comment]
@jczaja Approval CI still failed. but Xieyushun already approved but it still failed.
Coverage CI failed at paddle_bfloat package downloading. I rerun it. Let's make coverage CI pass. Do you know what is the new problem of approval CI?

**Comment by lidanqing-vv** [comment]
Hi, @XieYunshen Could you please approve this PR again? Cause last time of your approval is stale because @jczaja has to rebase newest develop branch. Thanks!

@baoachun Please help with the PR-CI-APPROVAL. This PR was approved before just @jczaja has to rebase newest develop to make all others pass because of out of 7 days. So the old APPROVAL was dismissed.

Thanks!

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_mkldnn_conv3d_op PROPERTIES TIMEOUT 300)`

**Pull request merged**